### DISCUSSION TITLE
Segmentation fault (core dumped)

### DISCUSSION THREAD

**Discussion by njzjz** [deepmodeling/deepmd-kit#7] [Q&A] [answered]
When I use `dp_train`, an error occurred:
> 2018-08-07 11:39:39.538273: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
> Segmentation fault (core dumped)

I don't know what went wrong.

**Comment by amcadmus** [comment]
Could you provide more details on your training?  what are your training parameters and data?

**Comment by njzjz** [comment]
Here are all files I used:
[C1111.tar.gz](https://github.com/deepmodeling/deepmd-kit/files/2265793/C1111.tar.gz)

I want to train complex systems containing different types of atoms. I used `conda install deepmd -c deepmd.kit -y` to install DeePMD. Then I converted raw data to numpy data and ran `dp_train C1111.json` command at the `train` directory.

**Comment by amcadmus** [comment, marked as answer]
For O, you used the axis rule [0, 0, 0, 0, 0, 1], which asks for at least two C(type 0) neighbours. Some of your molecules have only one carbon, so the dp_train failed to make the axes for them.

**Comment by amcadmus** [comment]
Btw, there is not virial in your data, please set the virial prefactors to 0.

**Comment by njzjz** [comment]
I have changed `axis_rule` to `[0,-2,0,0,-2,1,0,-2,0,0,-2,1,0,-2,0,0,-2,1]` but it still occurred even in the latest version. I analyzed core dump file with `gdb`:
```
Using host libthread_db library "/lib64/libthread_db.so.1".
Missing separate debuginfo for /opt/conda3/lib/python3.6/site-packages/numpy/../../../libiomp5.so
Try: yum --enablerepo='*debug*' install /usr/lib/debug/.build-id/50/126f8244a53ed88a1531546fcaa8dedc4bc85c.debug
Core was generated by `python3 /opt/deepmd/bin/dp_train ../train/C1111.json'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007f9c856c241c in compute_descriptor(std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> >&, std::vector<double, std::allocator<double> > const&, int const&, std::vector<int, std::allocator<int> > const&, SimulationRegion<double> const&, bool const&, int const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, int, int, int, int) ()
   from /opt/deepmd/lib/deepmd/libop_abi.so
Missing separate debuginfos, use: debuginfo-install glibc-2.17-260.el7.x86_64 libuuid-2.23.2-59.el7.x86_64
```
It seems the error occurred in [`compute_descriptor`](../blob/master/source/lib/include/ComputeDescriptor.h) but I have no idea about it.
